src/useGet.ts

```diff
@@ -94,26 +95,22 @@ export default <CustomApplication extends Application>(feathers: CustomApplicati
       isLoading.value = false;
     };
 
-    watch(_id, async () => {
-      await get();
-    });
-
-    const connectListener = () => {
+    const load = () => {
       void get();
     };
 
-    feathers.on('connect', connectListener);
+    const unload = () => {
+      unloadEventHandlers();
+      feathers.off('connect', load);
+    };
 
-    onMounted(async () => {
-      await get();
-    });
+    watch(_id, load, { immediate: true });
+    feathers.on('connect', load);
 
-    if (!disableUnloadingEventHandlers) {
-      onBeforeUnmount(() => {
-        unloadEventHandlers();
-        feathers.off('connect', connectListener);
-      });
+    // check if composition was called from inside a component setup function
+    if (getCurrentInstance()) {
+      onBeforeUnmount(unload);
     }
 
-    return { isLoading, data };
+    return { isLoading, data, load, unload };
```

Comment:
Should there be a test for `load` as it is external API?

Reply:
Dunno if it makes sense to export it at all as it will be called by the watcher. Maybe we should not export it for now.

Reply:
removed it for now